### DISCUSSION TITLE
Please add support the the RP Pico 2 W RP2350

### DISCUSSION THREAD

**Discussion by milelo** [pschatzmann/arduino-audio-tools#1816] [Q&A] [answered]
Please add support for the RP2350 or provide suitable settings. Thanks.

**Comment by pschatzmann** [comment, marked as answer]
What makes you think that it is not supported ?
It is my expectation that it should work...

**Reply by milelo** [reply]
My C compiler experience is limited. I'm trying this example with `#include <Arduino.h>` added:
https://github.com/pschatzmann/arduino-audio-tools/wiki/Encoding-and-Decoding-of-Audio#streaming-decoding

URLStream isn't defined. So it seems like AudioConfig.h isn't providing a suitable configuration.

platformio.ini:
```
[env:rpipico2w]
platform = https://github.com/maxgerhardt/platform-raspberrypi.git
board = rpipico2w
framework = arduino
board_build.core = earlephilhower
;board_build.filesystem_size: 0m, allocates all available flash memory for program code and data.
;& prevents the creation of a separate partition for a filesystem.
board_build.filesystem_size = 0m
debug_tool = cmsis-dap
;debug_speed kHz
debug_speed = 10000
upload_protocol = cmsis-dap
build_type = debug

monitor_speed = 115200
;Use "Serial" for RP Pico USB COM9 and "Serial1" for Pico Probe COM10 monitor.
monitor_port = COM10
monitor_filters = 
    default
    time

lib_deps =
    ;https://github.com/pschatzmann/arduino-audio-tools.git@0.9.8
    ;https://github.com/pschatzmann/arduino-libvorbis-tremor.git@1.0.2
    ;https://github.com/pschatzmann/arduino-libopus.git@1.3.1
    https://github.com/pschatzmann/arduino-audio-tools.git
    https://github.com/pschatzmann/arduino-libvorbis-tremor.git
    https://github.com/pschatzmann/arduino-libopus.git
    https://github.com/pschatzmann/arduino-libhelix.git
```

**Comment by pschatzmann** [comment]
The RP2350 does not have any wifi hardware support!
The Pico W has...

**Reply by milelo** [reply]
Sorry I don't understand your comment. The RP Pico 2 W supports WiFi. What are you referring to?

**Reply by pschatzmann** [reply]
Oh, I did not realize that this is out and I was just reading your description which was omitting this info.

What board would you select ?

With the Rasperry Pico W, a -DLWIP_IPV4=1 is added to the compilation commands to indicate that the board supports IP.

**Reply by pschatzmann** [reply]
I [committed some change](https://github.com/pschatzmann/arduino-audio-tools/commit/839eb4fee7faaf435c0bf3535f6c958ba8031bcf) that should work if you select "Generic RP2350"

I am not sure if PlatformIO is adding the same build options as Arduino automatically, so you might need to adjust the build_flags. Arduino for "Generic RP2350"  is using:

    build_flags = -DARDUINO_ARCH_RP2040 -DLWIP_IPV4=1


ps. dont forget to delete the cached libraries in PlatformIO, to trigger a reload of the latest update.

**Reply by milelo** [reply]
Great, thanks Phil. It builds and runs now with those flags.
